### ISSUE TITLE
Usage with Google Map

### ISSUE THREAD

**Issue by elenat82** [rubenCodeforges/ng-gapi#80]
In addition to integrating a Google Calendar in my app, I would like to integrate a Google Map as well, I tried following this tutorial (I am using Angular 7, maybe it could be a problem) : [https://medium.com/@balramchavan/integrating-google-maps-in-angular-5-ca5f68009f29](https://medium.com/@balramchavan/integrating-google-maps-in-angular-5-ca5f68009f29) but I have several problems:
1. where do I put my API Key? interface NgGapiClientConfig has no property for that 
2. `import { } from '@types/googlemaps';` this doesn't compile, error says: "File '.../node_modules/@types/googlemaps/index.d.ts' is not a module.
error TS6137: Cannot import type declaration files. Consider importing 'googlemaps' instead of '@types/googlemaps'." even if I ran `npm install --save-dev @types/googlemaps.`
I also tried: `/// <reference path="mypath/node_modules/@types/googlemaps/index.d.ts"/>` at the very top of my component file because someone here [https://stackoverflow.com/a/54445382/9113074](https://stackoverflow.com/a/54445382/9113074) was using Angular 7 and said it worked but it didn't, error is: `ERROR ReferenceError: google is not defined` at this line: `center: new google.maps.LatLng(18.5793, 73.8143),`
I also tried adding `var google: any;` as suggested here [https://medium.com/@gerhard.pazout/i-get-an-error-from-the-line-where-mapprop-is-declared-24b86047d86b](https://medium.com/@gerhard.pazout/i-get-an-error-from-the-line-where-mapprop-is-declared-24b86047d86b) but still no luck, and even this [https://medium.com/@carvarr/i-had-following-errors-in-my-angular-7-project-8eaf23bf565c](https://medium.com/@carvarr/i-had-following-errors-in-my-angular-7-project-8eaf23bf565c) didn't work.

Here is my component: 
/// `<reference path="../../../node_modules/@types/googlemaps/index.d.ts"/>`
`       
                import { Component, OnInit, ViewChild, ChangeDetectionStrategy, ViewEncapsulation, 
                ChangeDetectorRef, NgZone } from '@angular/core';
                import { GoogleAuthService } from 'ng-gapi';
                import { GoogleApiService } from 'ng-gapi';
                import GoogleUser = gapi.auth2.GoogleUser;
                import GoogleAuth = gapi.auth2.GoogleAuth;
                import { GoogleUserService } from '../services/google-user.service';

        @Component({
          selector: 'app-map',
          templateUrl: './map.component.html',
          styleUrls: ['./map.component.css'],
          changeDetection: ChangeDetectionStrategy.OnPush,
          encapsulation: ViewEncapsulation.None
        })
        export class MapComponent implements OnInit {
          @ViewChild('gmap') gmapElement: any;
          googleUser: GoogleUser;
          map: google.maps.Map;
          

          constructor(private cdRef: ChangeDetectorRef, private googleApiService: GoogleApiService, private ngZone: NgZone, private googleUserService: GoogleUserService, private googleAuthService: GoogleAuthService) { }

          ngOnInit() {

            this.googleApiService.onLoad().subscribe(
              (res) => {
                console.log("libreria Google API caricata!");
                if (this.googleUserService.isUserSignedInGoogle()) {
                  const googleToken = this.googleUserService.getGoogleToken();
                  var mapProp = {
                    center: new google.maps.LatLng(18.5793, 73.8143),
                    zoom: 15,
                    mapTypeId: google.maps.MapTypeId.ROADMAP
                  };
                  this.map = new google.maps.Map(this.gmapElement.nativeElement, mapProp);
                  
                }
                else {
                  console.log("l'utente non è ancora loggato");
                  
                  this.googleUserService.signInGoogle().subscribe(
                    (auth) => {
                      console.log("auth....", auth);
                      auth.signIn()
                        .then(
                          (res) => {
                            console.log("autenticazione a google andata a buon fine", res);
                            this.googleUserService.signInGoogleSuccessHandler(res);
                            const googleToken = this.googleUserService.getGoogleToken();
                            var mapProp = {
                              center: new google.maps.LatLng(18.5793, 73.8143),
                              zoom: 15,
                              mapTypeId: google.maps.MapTypeId.ROADMAP
                            };
                            this.map = new google.maps.Map(this.gmapElement.nativeElement, mapProp);
                            
                          },
                          (err) => {
                            console.log("autenticazione a google andata male");
                            this.googleUserService.signInGoogleErrorHandler(err)
                          })
                    },
                    (err) => { console.log("errore nella funzione googleUserService.signInGoogle()", err) }
                  );
                }
              },
              (err) => { console.log("errore nel caricamento della libreria Google API ->", err) }
            );

            
          }

        }
`

**Comment by rubenCodeforges** [comment]
Well ng-gapi has nothing to do with google maps `JavaScript API`.

In your case you need to follow the tutorial. 
Just use the step by steps instructions:

```
Step 3: Link Google Maps JavaScript CDN inside index.html

<script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&callback=initMap"
    async defer></script>
```

As to the `import { } from '@types/googlemaps'; this doesn't compile, error says: "File ` - this is because you are using typings wrong, they have no logic , they are just a representation of the js code you import with `<script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&callback=initMap"
    async defer></script>` in typescript.

 In other words this `/// <reference path="../../../node_modules/@types/googlemaps/index.d.ts"/>` is wrong use `import`

**Comment by rubenCodeforges** [comment]
Also i would recommend always to check the official docs https://developers.google.com/maps/documentation/javascript/tutorial here you can see that maps is not a REST API

**Comment by elenat82** [comment]
Sorry, I thought ng-gapi would have loaded all Google libraries and didn't want to load the library twice.
My fault, you can close this.
Thank you

**Comment by rubenCodeforges** [comment]
@elenat82 no problem , just refer to the official gapi docs , there you can see what it supports
